pkg/azuredisk/controllerserver.go

```diff
@@ -650,6 +650,14 @@ func generateCSISnapshot(sourceVolumeID string, snapshot *compute.Snapshot) (*cs
 		return nil, fmt.Errorf("diskSizeGB of snapshot property is nil")
 	}
 
+	if sourceVolumeID == "" {
+		if snapshot.SnapshotProperties != nil &&
```

Comment:
`snapshot.SnapshotProperties != nil` is already checked by L639.

Reply:
done.